wp_api/src/api_client.rs

```diff
@@ -164,6 +146,18 @@ impl WpApiClient {
     }
 }
 
+// IMPORTANT: This type may be aggressively cloned, so all of its fields must be cheap to clone!
+#[derive(Clone, uniffi::Record)]
+pub struct WpApiClientDelegate {
```

Comment:
This is more of a data source object than a delegate – I'd expect a delegate to be more like #693 (where it says "this thing happened, and you need to do something about it") instead of a set of configuration parameters.

I'd propose changing this to  `WpApiClientDataSource` or `WpApiClientConfiguration`

Reply:
We picked this name together on Slack 😅 I like configuration more, because even though it's not directly a delegate, it still sort of acts like one because it contains `request_executor`.

Regardless, I don't feel strongly about any name, so if you pick the name, I am happy to open a new PR to update it. However, I'll probably do it after #693 lands to avoid unnecessary conflicts.